src/client_.rs

```diff
@@ -518,33 +512,36 @@ impl Part {
     ///
     fn new<N, F>(inner: Inner, name: N, mime: Option<Mime>, filename: Option<F>) -> Part
     where
-        N: Into<String>,
-        F: Into<String>,
+        N: Display,
+        F: Display,
     {
         // `name` disposition parameter is required. It should correspond to the
         // name of a form field.
         //
         // [See 4.2](https://tools.ietf.org/html/rfc7578#section-4.2)
         //
-        let mut disposition_params = vec![DispositionParam::Ext("name".into(), name.into())];
+        let mut disposition_params = vec![format!("name=\"{}\"", name)];
 
         // `filename` can be supplied for files, but is totally optional.
         //
         // [See 4.2](https://tools.ietf.org/html/rfc7578#section-4.2)
         //
         if let Some(filename) = filename {
-            disposition_params.push(DispositionParam::Ext("filename".into(), filename.into()));
+            disposition_params.push(format!("filename=\"{}\"", filename));
         }
 
-        let content_type = ContentType(mime.unwrap_or_else(|| inner.default_content_type()));
+        let content_type = format!(
+            "Content-Type: {}",
+            mime.unwrap_or_else(|| inner.default_content_type())
+        );
 
         Part {
             inner: inner,
             content_type: content_type,
-            content_disposition: ContentDisposition {
-                disposition: DispositionType::Ext("form-data".into()),
-                parameters: disposition_params,
-            },
+            content_disposition: format!(
+                "Content-Disposition: form-data; {}",
+                disposition_params.join("; ")
+            ),
```

Comment:
As opposed to having the header name be in the variable for `content_disposition` and `content_type`, what do you think about just writing the header names in the `write_headers` function?

Also a little bit disappointed that those types were removed.